src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +95,86 @@ def __init__(self, proto_unix_io):
         self.geneva_value = 3
         self.power_value = 1
 
+    def kick_clicked(self):
+        # if button is enabled
+        if self.kick_button_enable:
+            # send kick primitive
+            self.send_command(ChickerCommandMode.KICK)
+            self.toggle_kick_button()
+
+            # set and start timer to re-enable kick button after 3 seconds
+            self.start_timer_once(self.toggle_kick_button, 3 * MILLISECONDS_PER_SECOND)
+
+    def chip_clicked(self):
+        # if button is enabled
+        if self.chip_button_enable:
```

Comment:
I know that some of these functions are pretty straight forward, but to stay consistent with formatting in the rest of the code base, we should add a simple comment describing the function. (optional) In the scenario of the callback functions, triggered based on an event, we could note that in the comment.

Reply:
accidentally requested a review, i'm working on these now

Reply:
Message me when it's ready! :smiley:

Reply:
done